smithy-model/src/test/java/software/amazon/smithy/model/transform/RenameShapesTest.java

```diff
@@ -0,0 +1,242 @@
+        ModelTransformer transformer = ModelTransformer.create();
+        Map<ShapeId, ShapeId> renamed = new HashMap<ShapeId, ShapeId>();
+        Model result = transformer.renameShapes(model, renamed);
+        assertThat(result.shapes().count(), Matchers.equalTo(1L));
+        assertThat(result.getShape(stringId).get(), Matchers.is(fooTarget));
+    }
+
+    @Test
+    public void returnsUnmodifiedModelIfToAndFromAreEqual() {
+        ShapeId stringId = ShapeId.from("ns.foo#String");
+        StringShape target = StringShape.builder().id(stringId).build();
+        Model model = Model.builder()
+                .addShapes(target)
+                .build();
+        ModelTransformer transformer = ModelTransformer.create();
+        Map<ShapeId, ShapeId> renamed = new HashMap<ShapeId, ShapeId>() {
+            { put(stringId, stringId ); }
+        };
+        Model result = transformer.renameShapes(model, renamed);
+        assertThat(result.shapes().count(), Matchers.equalTo(1L));
+        assertThat(result.getShape(stringId).get(), Matchers.is(target));
+    }
+
+    @Test
+    public void returnsModelWithRenamedStringShape() {
+        ShapeId fromStringId = ShapeId.from("ns.foo#String");
+        ShapeId containerId = ShapeId.from("ns.foo#Container");
+        ShapeId keyId = ShapeId.from("ns.foo#Container$key");
+        ShapeId valueId = ShapeId.from("ns.foo#Container$value");
+
+        StringShape target = StringShape.builder().id(fromStringId).build();
+        MemberShape keyMember = MemberShape.builder().id(keyId).target(fromStringId).build();
+        MemberShape valueMember = MemberShape.builder().id(valueId).target(fromStringId).build();
+        MapShape container = MapShape.builder().id(containerId).key(keyMember).value(valueMember).build();
+        Model model = Model.builder()
+                .addShapes(target, keyMember, valueMember, container)
+                .build();
+        ModelTransformer transformer = ModelTransformer.create();
+
+        ShapeId toStringId = ShapeId.from("ns.bar#String");
+        Map<ShapeId, ShapeId> renamed = new HashMap<ShapeId, ShapeId>() {{
+           put(fromStringId, toStringId);
+        }};
+        Model result = transformer.renameShapes(model, renamed);
+
+        assertThat(result.shapes().count(), Matchers.equalTo(4L));
+        assertThat(result.getShape(toStringId).isPresent(), Matchers.is(true));
+        assertThat(result.getShape(fromStringId).isPresent(), Matchers.is(false));
+    }
+
+    @Test
+    public void updatesIdRefsWhenValueRenamed() {
+        Model model = Model.assembler()
+                .addImport(IntegTest.class.getResource("rename-shape-test-model.json"))
+                .assemble()
+                .unwrap();
+
+        ShapeId fromId = ShapeId.from("ns.foo#OldShape");
+        ShapeId toId = ShapeId.from("ns.foo#NewShape");
+        Map<ShapeId, ShapeId> renamed = new HashMap<ShapeId, ShapeId>() {
+            { put(fromId, toId ); }
```

Comment:
nit: This is valid Java, but pretty alien in practice. I'd just use a normal put.